### PR TITLE
PowerModel Data Text Summary

### PR DESCRIPTION
**Pull request by ccoffrin**
@yeesian, suggestions welcome.

### CODE REVIEW

**Comment by yeesian** [comment]
Might be a good idea to print it to an IO object instead? Here's [an example](https://github.com/IainNZ/JuMPeR.jl/blob/090ce1dc0a9c851fe97caedf08ab004fabaf705c/src/print.jl#L36-L63). If you don't mind, I can push a commit to this branch tonight.

**Comment by ccoffrin** [comment]
Yes absolutely, a PR is welcome.

**Comment by ccoffrin** [comment]
Discussion Notes
- [ ] add output to IO object
- [ ] add version that reads data from a path as a string and returns a dict
- [ ] make work with solution objects
- [ ] grey text does not render the same on windows cmd
- [ ] bold text does not render the same on windows power shell

**Comment by yeesian** [comment]
Nice, I had a local branch going, but you've gotten to it in https://github.com/lanl-ansi/PowerModels.jl/pull/240/commits/3be75f5aa3f62ef110ecda11e491f0cf55c330ff alr 👍

**Comment by ccoffrin** [comment]
Sorry!  I already started feature adds so it was incremental additional effort.

**Comment by yeesian** [comment]
Yeah it's all good :) Happy to see things moving along!

File: test/data.jl [hunk 1]
```diff
@@ -1,5 +1,41 @@
 # Tests of data checking and transformation code
 
+@testset "test data summary" begin
+    buf = IOBuffer()
+
+    @testset "5-bus summary from dict" begin
+        data = PowerModels.parse_file("../test/data/matpower/case5.m")
+        PowerModels.summary(data, io=buf)
+
+        output = String(take!(buf))
```
Comment (yeesian) [inline test/data.jl]: `output = sprint((io,d) -> PowerModels.summary(d, io=io), data)` to replace L7-10 is also possible.

File: test/data.jl [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+        @test contains(output, "Table: load")
+        @test contains(output, "Table: gen")
+        @test contains(output, "Table: branch")
+        @test contains(output, "Table: areas")
+    end
+
+    @testset "5-bus summary from file location" begin
+        data = PowerModels.summary("../test/data/matpower/case5.m", io=buf)
+
+        output = String(take!(buf))
```
Comment (yeesian) [inline test/data.jl]: similarly here: `output = sprint((io,filename) -> PowerModels.summary(filename, io=io), "../test/data/matpower/case5.m")`

if you want a `IOBuffer`-free way of doing it.

File: test/data.jl (hunk 1 above)
Comment (yeesian) [reply test/data.jl]: An alternative is to implement

```julia
summary(io::IO, data::Dict)
    ....
end

print_summary(data::Any)
    summary(STDOUT, data)
end
```

Then this line will be `output = sprint(PowerModels.summary, data)` instead.

Comment (ccoffrin) [reply test/data.jl]: If you do change the function name, make sure to update the data doc md to be consistent.

**Pull request merged**